tests/test_DataSet.py

```diff
@@ -0,0 +1,59 @@
+import numpy as np
+import pytest
+import os
+
+import RadClass.DataSet as ds
+import tests.test_data as test_data
+
+# randomized data to store (smaller than an actual MUSE file)
+live = np.random.rand(test_data.timesteps,)
+timestamps = np.random.rand(test_data.timesteps,)
+spectra = np.random.rand(test_data.timesteps, test_data.energy_bins)
+
+
+@pytest.fixture(scope="session", autouse=True)
+def init_test_file():
+    yield test_data.create_file(live, timestamps, spectra)
+    os.remove(test_data.filename)
```

Comment:
@gonuke This smells like bad practice to me, but only one of these `os.remove` statements is needed because both tests (`test_DataSet` and `test_RadClass`) use the same `test_data.filename`. In fact, if I included an `os.remove` in the second feature (in `test_RadClass.py`), then an error would be thrown, since the file with the name `test_data.filename` was removed by the first fixutre (in `test_DataSet.py`).

If I understand correctly, this is because even though the files are created twice, once by each fixture, the cleanup is done sequentially at the end of the pytest execution. I did confirm locally that two _separate_ files are created and used by the tests. Is this sufficient or should I approach cleanup in a different way (by possible using different filenames)?

Reply:
Interesting... maybe only define this method once like described here: https://www.tutorialspoint.com/pytest/pytest_conftest_py.htm

Reply:
The only problem with defining it once is that the data written to file by each test "group" is subtely different. I guess I could make the data written consistent test-to-test instead. Or, I could write two features that create two separate files with two separate names.

Reply:
@gonuke I think I solved it. I changed the scope of each fixture. Rather than having `scope='session'`, I have changed each feature to have `scope='module'`. I think this restricts the scope of the fixture to use within each individual test file (thereby limited it to the respective `test_DataSet` or `test_RadClass`). In combination with `autouse=True`, each set of tests has its requisite test data file that is removed at the end of the testing.